### DISCUSSION TITLE
Proposal for Simulator Pre-selection via Command-Line Flag in Tauri's Development Environment

### DISCUSSION THREAD

**Discussion by nyxb** [tauri-apps/tauri#9664] [General]
Hello Tauri Community,

I am currently working on a Tauri application within a monorepo setup, utilizing Turbo to manage multiple packages including desktop, iOS, and Android versions of our application. One challenge I've encountered is the inability to simultaneously launch multiple device simulations for iOS and Android due to the required manual selection of simulators at runtime. This issue complicates the workflow significantly, especially when aiming to develop and test across multiple platforms concurrently.

To enhance developer productivity and integration with tools like Turbo in monorepos, I propose the addition of a feature to Tauri's CLI: the ability to pre-select simulators/devices via a command-line flag. For example, a developer could specify the simulator for an iOS build directly in the command like this:

```bash
tauri dev ios --simulator "iPhone 15 Pro Max"
```

This flag would allow the CLI to bypass the interactive simulator selection menu, making it more script-friendly and conducive to automated workflows. It would be particularly beneficial in environments where developers need to see real-time interactions across desktop, iOS, and Android without manually starting each simulator.

I believe this feature would not only streamline the development process for individual developers but also enhance the efficiency of teams working in complex app ecosystems.

Looking forward to hearing your thoughts and suggestions on this proposal.

Best regards,
nyxb

**Comment by FabianLars** [comment]
I'd have to double check when I have access to my computer tomorrow but I thought this was already a thing but without the flag:
```
tauri ios dev "iPhone 15 Pro Max"
```

If I'm wrong then I agree that this would be a valuable addition :)

**Reply by justinmoon** [reply]
This does work
